### ISSUE TITLE
Keras Reshae layer doesn't support in tf 2.2 ?

### ISSUE THREAD

**Issue by cuge1995** [tensorflow/tensorflow#40445]

**Comment by cuge1995** [comment]
Reshape layer

**Comment by Saduf2019** [comment]
@cuge1995 
Please share simple stand alone code where issue is faced while using the reshape layer for us to analyse the issue faced.

**Comment by cuge1995** [comment]
# input_Transformation_net
input_points = Input(shape=(num_points, 3))
x = Convolution1D(64, 1, activation='relu',
                  input_shape=(num_points, 3))(input_points)
x = BatchNormalization()(x)
x = Convolution1D(128, 1, activation='relu')(x)
x = BatchNormalization()(x)
x = Convolution1D(1024, 1, activation='relu')(x)
x = BatchNormalization()(x)
x = MaxPooling1D(pool_size=num_points)(x)
x = Dense(512, activation='relu')(x)
x = BatchNormalization()(x)
x = Dense(256, activation='relu')(x)
x = BatchNormalization()(x)
x = Dense(9, weights=[np.zeros([256, 9]), np.array([1, 0, 0, 0, 1, 0, 0, 0, 1]).astype(np.float32)])(x)
input_T = Reshape((3, 3))(x)

# forward net
g = Lambda(mat_mul, arguments={'B': input_T})(input_points)
g = Convolution1D(64, 1, input_shape=(num_points, 3), activation='relu')(g)
g = BatchNormalization()(g)
g = Convolution1D(64, 1, input_shape=(num_points, 3), activation='relu')(g)
g = BatchNormalization()(g)


such keras code not running in tf 2.2

**Comment by Saduf2019** [comment]
@cuge1995 
I ran the code shared above and face a different error, please share all dependencies and executable code for us to replicate.
Please find the [gist here](https://colab.research.google.com/gist/Saduf2019/57204907eb11a520ea77b89b01bdda2a/untitled231.ipynb)

**Comment by cuge1995** [comment]
import numpy as np
import os
import tensorflow as tf
from keras import optimizers
from keras.layers import Input
from keras.models import Model
from keras.layers import Dense, Flatten, Reshape, Dropout
from keras.layers import Convolution1D, MaxPooling1D, BatchNormalization
from keras.layers import Lambda
from keras.utils import np_utils
import h5py


def mat_mul(A, B):
    return tf.matmul(A, B)


def load_h5(h5_filename):
    f = h5py.File(h5_filename)
    data = f['data'][:]
    label = f['label'][:]
    return (data, label)


def rotate_point_cloud(batch_data):
    """ Randomly rotate the point clouds to augument the dataset
        rotation is per shape based along up direction
        Input:
          BxNx3 array, original batch of point clouds
        Return:
          BxNx3 array, rotated batch of point clouds
    """
    rotated_data = np.zeros(batch_data.shape, dtype=np.float32)
    for k in range(batch_data.shape[0]):
        rotation_angle = np.random.uniform() * 2 * np.pi
        cosval = np.cos(rotation_angle)
        sinval = np.sin(rotation_angle)
        rotation_matrix = np.array([[cosval, 0, sinval],
                                    [0, 1, 0],
                                    [-sinval, 0, cosval]])
        shape_pc = batch_data[k, ...]
        rotated_data[k, ...] = np.dot(shape_pc.reshape((-1, 3)), rotation_matrix)
    return rotated_data


def jitter_point_cloud(batch_data, sigma=0.01, clip=0.05):
    """ Randomly jitter points. jittering is per point.
        Input:
          BxNx3 array, original batch of point clouds
        Return:
          BxNx3 array, jittered batch of point clouds
    """
    B, N, C = batch_data.shape
    assert(clip > 0)
    jittered_data = np.clip(sigma * np.random.randn(B, N, C), -1 * clip, clip)
    jittered_data += batch_data
    return jittered_data



num_points = 2048


k = 40


adam = optimizers.Adam(lr=0.001, decay=0.7)


input_points = Input(shape=(num_points, 3))
x = Convolution1D(64, 1, activation='relu',
                  input_shape=(num_points, 3))(input_points)
x = BatchNormalization()(x)
x = Convolution1D(128, 1, activation='relu')(x)
x = BatchNormalization()(x)
x = Convolution1D(1024, 1, activation='relu')(x)
x = BatchNormalization()(x)
x = MaxPooling1D(pool_size=num_points)(x)
x = Dense(512, activation='relu')(x)
x = BatchNormalization()(x)
x = Dense(256, activation='relu')(x)
x = BatchNormalization()(x)
x = Dense(9, weights=[np.zeros([256, 9]), np.array([1, 0, 0, 0, 1, 0, 0, 0, 1]).astype(np.float32)])(x)
input_T = Reshape((3, 3))(x)

g = Lambda(mat_mul, arguments={'B': input_T})(input_points)
g = Convolution1D(64, 1, input_shape=(num_points, 3), activation='relu')(g)
g = BatchNormalization()(g)
g = Convolution1D(64, 1, input_shape=(num_points, 3), activation='relu')(g)
g = BatchNormalization()(g)


f = Convolution1D(64, 1, activation='relu')(g)
f = BatchNormalization()(f)
f = Convolution1D(128, 1, activation='relu')(f)
f = BatchNormalization()(f)
f = Convolution1D(1024, 1, activation='relu')(f)
f = BatchNormalization()(f)
f = MaxPooling1D(pool_size=num_points)(f)
f = Dense(512, activation='relu')(f)
f = BatchNormalization()(f)
f = Dense(256, activation='relu')(f)
f = BatchNormalization()(f)
f = Dense(64 * 64, weights=[np.zeros([256, 64 * 64]), np.eye(64).flatten().astype(np.float32)])(f)
feature_T = Reshape((64, 64))(f)

g = Lambda(mat_mul, arguments={'B': feature_T})(g)
g = Convolution1D(64, 1, activation='relu')(g)
g = BatchNormalization()(g)
g = Convolution1D(128, 1, activation='relu')(g)
g = BatchNormalization()(g)
g = Convolution1D(1024, 1, activation='relu')(g)
g = BatchNormalization()(g)

global_feature = MaxPooling1D(pool_size=num_points)(g)


c = Dense(512, activation='relu')(global_feature)
c = BatchNormalization()(c)
c = Dropout(rate=0.7)(c)
c = Dense(256, activation='relu')(c)
c = BatchNormalization()(c)
c = Dropout(rate=0.7)(c)
c = Dense(k, activation='softmax')(c)
prediction = Flatten()(c)

model = Model(inputs=input_points, outputs=prediction)
print(model.summary())

**Comment by Saduf2019** [comment]
@
I ran the above code and fixed as many indentation and syntax error as i could but there are way to many to fix, please find the [gist here](https://colab.research.google.com/gist/Saduf2019/f1f7701ca7f4db0ad099d2aae2bb38b8/untitled231.ipynb).
Please provide executable code such that it replicates the issue faced or if possible please share a colab gist for us to analyse.

**Comment by cuge1995** [comment]
https://github.com/garyli1019/pointnet-keras

**Comment by cuge1995** [comment]
the train_cls.py file

**Comment by cuge1995** [comment]
I can build the model just using keras, but not work when I using tf.keras

**Comment by jvishnuvardhan** [comment]
@cuge1995 Can you please update your code in the [colab gist](https://colab.research.google.com/gist/jvishnuvardhan/c5e3ce3fb15aa0a1bba4efbdf36d44e7/-40445.ipynb) I prepared from your repo. Next time, please try to provide a standalone code so that the resolution will be faster. Thanks!

**Comment by cuge1995** [comment]
thanks, but this model still cannot run


_SymbolicException: Inputs to eager execution function cannot be Keras symbolic tensors, but found [<tf.Tensor 'reshape_2/Identity:0' shape=(None, 3, 3) dtype=float32>, <tf.Tensor 'reshape_3/Identity:0' shape=(None, 64, 64) dtype=float32>]

**Comment by cuge1995** [comment]
you can look at this kaggle kernel, which contain the dataset
https://www.kaggle.com/cuge1995/tpu-2-transformers

**Comment by cuge1995** [comment]
??

**Comment by jvishnuvardhan** [comment]
@cuge1995 Can you please update the colab or create a simple standalone code to reproduce the issue? Kaggle's kernel link throws '404' error. But, please provide a standalone code to to reproduce the issue. You could use any public data. Thanks!
